src/Symfony/Component/Serializer/Encoder/CsvEncoder.php

```diff
@@ -189,11 +189,38 @@ private function flatten(array $array, array &$result, $keySeparator, $parentKey
             if (\is_array($value)) {
                 $this->flatten($value, $result, $keySeparator, $parentKey.$key.$keySeparator);
             } else {
-                $result[$parentKey.$key] = $value;
+                $result[$parentKey.$key] = $this->cleanValueByType($value);
             }
         }
     }
 
+    /**
+     * Ensure an actual value is used instead of a blank value when dealing
```

Comment:
```suggestion
     * Ensures an actual value is used instead of a blank value when dealing
```

Reply:
Thanks for your review @dunglas :)
Done, and also fixed up the formatting errors listed by fabbot.io.